fendermint/vm/topdown/src/sync/pointers.rs

```diff
@@ -31,6 +31,11 @@ impl SyncPointers {
     pub fn set_tail(&mut self, height: BlockHeight, hash: BlockHash) {
         self.tail = Some((height, hash));
     }
+
+    pub fn reset(&mut self, head: BlockHeight) {
+        self.tail = None;
+        self.head = head;
```

Comment:
As discussed sync, I think with the new model where we enforced the defer execution of side-effects for top-down there is no need to use a `head` and a `tail` here.